avalanche/models/ncm_classifier.py

```diff
@@ -10,22 +10,82 @@ class NCMClassifier(nn.Module):
     ones stored in 'self.class_means'.
     """
 
-    def __init__(self, class_mean=None, normalize=True):
+    def __init__(self, class_means_dict={}, normalize=True):
         """
-        :param class_mean: tensor of dimension (num_classes x feature_size)
-            used to classify input patterns.
+        :param class_means_dict: dictionary mapping class_id to mean vector.
+            classes must be zero-indexed.
         :param normalize: whether to normalize the input with
             2-norm = 1 before computing the distance.
         """
         super().__init__()
-        self.class_means = class_mean
+        assert isinstance(class_means_dict, dict), \
+            "class_means_dict must be a dictionary mapping class_id " \
+            "to mean vector"
+        self.class_means_dict = class_means_dict
```

Comment:
tensors in `class_means_dict` should be registered as buffers

Reply:
I can do that. The dictionary itself cannot be registered though. Is this a problem for check-pointing? 
In case, I guess I can implement a method that fills the dictionary with the current buffers after each NCM update.

Reply:
> Is this a problem for check-pointing?

I'm not sure, maybe add a test that uses `load_state_dict` and `save_state_dict` to make sure buffers are detected correctly.

Reply:
I had to override the default load but now loading/saving works.

Reply:
can you add a test just for safety?

Reply:
Done